File: lokesh.java
import java.util.Scanner;
import java.math.BigInteger;
import java.io.*;
import java.util.*;
import java.math.BigDecimal;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
      /*  if (System.getProperty("ONLINE_JUDGE") == null) {
            try {
                System.setOut(new PrintStream(
                    new FileOutputStream("output.txt")));
                sc = new Scanner(new File("input.txt"));
            }
            catch (Exception e) {
            }
            long startTime = System.nanoTime();
            long endTime = System.nanoTime();
            System.out.println("Took "+(endTime - startTime) + " ns"); 
        } */
       // int t = sc.nextInt();
        BigDecimal a = new BigDecimal(2.71828);
        a = a.pow(100000005,100000007);
        System.out.println(a);
    }
}

File: lokesh.py
def f(n,s,m):
    if s == 0:
        return 1;
    p=f(n,s//2,m)
    if s%2==0:
        return p*p%m;
    else:
        return n*p*p%m;





for _ in range(int(input())):
    a=list(map(int,input().split(" ")))
    n=a[0]
    p=str(a[1])
    m=a[2]
    q=0;
    for i in p:
        q*=3;
        q+=int(i);
    print(f(n,q,m))